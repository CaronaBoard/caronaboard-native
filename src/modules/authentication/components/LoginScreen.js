import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components/index'
import { styles } from '../containers/styles/LoginScreenStyles'

export const INITIAL_STATE = {
  email: '',
  password: '',
  loading: false
}

export default class LoginV2 extends React.Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  onButtonPress = () => {
    const { email, password } = this.state
    const { onButtonPress } = this.props
    try {
      onButtonPress(email, password)
    } catch (error) {
      console.log(error)
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/logo.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput onChangeText={(email) => { this.setState({email}) }} rkType='rounded' placeholder='Username' autoCapitalize='none' />
        <RkTextInput onChangeText={(password) => { this.setState({password}) }} rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton
            style={styles.loginButton}
            rkType='large'
            text='LOGIN'
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer()}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
