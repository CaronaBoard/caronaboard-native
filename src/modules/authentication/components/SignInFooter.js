import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
