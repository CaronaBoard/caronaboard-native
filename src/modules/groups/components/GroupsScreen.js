import React from 'react'
import { RkText } from 'react-native-ui-kitten'
import {
  ScrollView,
  View,
  TouchableOpacity
} from 'react-native'
import PropTypes from 'prop-types'

import { styles } from './styles/GroupsScreen.style'
import { GroupPropType } from '../types'
import { WarningScreen } from '../../shared/components/WarningScreen.style'

const hardCodedGroup = {
  'id': 'twpoa',
  'members': {},
  'name': 'ThoughtWorks POA'
}

export class GroupsScreen extends React.Component {
  static propTypes = {
    uid: PropTypes.string.isRequired,
    groups: PropTypes.arrayOf(GroupPropType).isRequired,
    joinGroup: PropTypes.func.isRequired,
    changeGroup: PropTypes.func.isRequired,
    activeGroup: PropTypes.oneOfType([GroupPropType, {}])
  }

  renderActiveGroup = (title, activeGroup) => {
    if (!activeGroup.name) {
      return null
    }

    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        <View style={styles.row} >
          <View
            style={styles.rowButton}
          >
            <RkText rkType='header6'>{activeGroup.name}</RkText>
          </View>
        </View>
      </View>
    )
  }

  renderGroupList = (title, group, onPress = () => null) => {
    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        {group.map(group => {
          return (
            <View style={styles.row} key={group.id}>
              <TouchableOpacity
                style={styles.rowButton}
                onPress={() => onPress(group)}
              >
                <RkText rkType='header6'>{group.name}</RkText>
              </TouchableOpacity>
            </View>
          )
        })}
      </View>
    )
  }

  render () {
    const {groups, uid, joinGroup, changeGroup, activeGroup} = this.props
    const groupsYouBelong = [hardCodedGroup]

    groups.forEach(group => {
      if (group.members.hasOwnProperty(uid)) {
        groupsYouBelong.push(group)
      }
    })

    if (!uid) {
      return (
        <WarningScreen activeGroup={'Must have profile first'} />
      )
    }

    return (
      <ScrollView style={styles.container}>
        {this.renderActiveGroup('GRUPO ATIVO', activeGroup)}
        {this.renderGroupList('GRUPOS QUE VOCE PERTENCE', groupsYouBelong, changeGroup)}
        {this.renderGroupList('TODOS OS GRUPOS', groups, joinGroup)}
      </ScrollView>
    )
  }
}
