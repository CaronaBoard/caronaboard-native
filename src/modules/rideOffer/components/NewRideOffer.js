import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { RkText } from 'react-native-ui-kitten'

import styles from './styles/YourRideOffer.style'
import { TextInput } from '../../shared/components/index'
import { LoadingSpinnerView } from '../../shared/components/LoadingSpinnerView'
import { GradientButton } from '../../shared/components/GradientButton'

export const INITIAL_STATE = {
  origin: '',
  destination: '',
  days: '',
  hours: '',
  loading: false
}

export class NewRideOffer extends React.PureComponent {
  state = INITIAL_STATE

  static propTypes = {
    onPress: PropTypes.func.isRequired
  }

  onPress = async () => {
    this.setState({loading: true})
    const { origin, destination, days, hours } = this.state
    await this.props.onPress({ origin, destination, days, hours })
    this.setState({loading: false})
  }

  render () {
    return (
      <LoadingSpinnerView isLoading={this.state.loading} style={styles.flexible}>
        <View style={styles.container}>
          <RkText style={styles.title}>
            Ride Offer
          </RkText>
          <View style={styles.inputTextsContainer}>
            <TextInput placeholder='ORIGEM' onChangeText={(origin) => this.setState({origin})} />
            <TextInput placeholder='DESTINO' onChangeText={(destination) => this.setState({destination})} />
            <TextInput placeholder='DIAS' onChangeText={(days) => this.setState({days})} />
            <TextInput placeholder='HORAS' onChangeText={(hours) => this.setState({hours})} />
            <View style={styles.centralized}>
              <GradientButton
                rkType='large'
                text={'Oferecer carona'}
                onPress={this.onPress}
              />
            </View>
          </View>
        </View>
      </LoadingSpinnerView>
    )
  }
}
