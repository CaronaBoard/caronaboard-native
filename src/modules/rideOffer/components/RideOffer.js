import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import Styles from './styles/RideOfferStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { RkCard } from 'react-native-ui-kitten'
import { ridePropTypes } from '../../rideRequest/types'

const renderLine = (text, icon) => {
  return (
    <View style={Styles.line} >
      <Icon name={icon} style={Styles.icon} />
      <Text rkCardText>{text}</Text>
    </View>
  )
}

export const RideOffer = props => {
  const {
    origin,
    destination,
    days,
    hours
  } = props.ride

  return (
    <TouchableOpacity>
      <RkCard>
        <View rkCardContent>
          { renderLine(origin, 'radio-button-unchecked') }
          { renderLine(destination, 'radio-button-unchecked') }
        </View>
        <View rkCardContent>
          { renderLine(days, 'today') }
          { renderLine(hours, 'schedule') }
        </View>
      </RkCard>
    </TouchableOpacity>
  )
}

RideOffer.propTypes = {
  ride: ridePropTypes
}
