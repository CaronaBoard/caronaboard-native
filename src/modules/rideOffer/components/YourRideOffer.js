import React from 'react'
import i18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { View, Text, TouchableOpacity } from 'react-native'

import Styles from './styles/YourRideOffer.style'
import { RkCard } from 'react-native-ui-kitten'
import { RidePropType } from '../../rideRequest/types'

export class YourRideOffer extends React.Component {
  static propTypes = {
    ride: RidePropType.isRequired,
    onPress: PropTypes.func.isRequired
  }

  onPress = () => {
    this.props.onPress(this.props.ride)
  }

  render () {
    const {
      origin,
      destination,
      days,
      hours
    } = this.props.ride

    return (
      <TouchableOpacity onPress={() => this.onPress()}>
        <RkCard>
          <View rkCardContent>
            <View style={Styles.line}>
              <Text rkCardText>
                {i18n.t('OFFERING_RIDE')}
              </Text>
              <View style={Styles.line}>
                <Text rkCardText>
                  {`De: ${origin}`}
                </Text>
              </View>
              <Text rkCardText>
                {`Para: ${destination}`}
              </Text>
            </View>
            <Text rkCardText>
              {`Em: ${days}`}
            </Text>
            <View style={Styles.line}>
              <Text rkCardText>
                {`As: ${hours}`}
              </Text>
            </View>
          </View>
          <Icon
            name='delete-forever'
            size={30}
            color='#900'
            style={Styles.deleteIcon}
          />
        </RkCard>
      </TouchableOpacity>
    )
  }
}
