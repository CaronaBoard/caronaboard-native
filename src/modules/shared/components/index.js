export { Button } from './Button'
export { TextInput } from './TextInput'
export { GradientButton } from './GradientButton'
export { LinearGradient } from './LinearGradient'
