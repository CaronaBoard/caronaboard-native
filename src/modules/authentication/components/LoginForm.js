import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton, TextInput } from '../../shared/components'
import { styles } from '../containers/styles/LoginScreenStyles'

export class LoginForm extends React.Component {
  static defaultProps = {
    footer: () => {},
    toast: {
      showAlert: false,
      message: ''
    }
  }

  state = {
    email: '',
    password: '',
    loading: false
  }

  onButtonPress = () => {
    const { email, password } = this.state
    this.props.onButtonPress(email, password)
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.toast

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <TextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <TextInput
          onChangeText={(password) => { this.setState({password}) }}
          key='password-input'
          rkType='rounded'
          placeholder='Password'
          secureTextEntry
        />

        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text={this.props.buttonText}
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer(this.state.email)}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
