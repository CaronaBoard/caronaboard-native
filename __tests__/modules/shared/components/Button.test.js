import React from 'react'
import { shallow } from 'enzyme'
import { RkButton } from 'react-native-ui-kitten'
import { Button } from '../../../../src/modules/shared/components'

const props = {
  text: 'texto',
  onPress: jest.fn()
}

describe('<Button />', () => {
  const wrapper = shallow(<Button {...props} />)

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call callback onPress', () => {
    wrapper.simulate('press')
    expect(props.onPress).toHaveBeenCalled()
  })

  it('Should have button text', () => {
    expect(wrapper.find(RkButton).prop('children')).toEqual(props.text)
  })
})
