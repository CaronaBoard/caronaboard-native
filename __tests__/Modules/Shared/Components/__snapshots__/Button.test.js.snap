// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should have a snapshot 1`] = `
<RkButton
  innerStyle={
    Array [
      Object {
        "fontSize": 20,
      },
    ]
  }
  onPress={[Function]}
  rkType="circle outline medium"
>
  texto
</RkButton>
`;
